server/service/packs.go

```diff
@@ -45,3 +87,446 @@ func (svc *Service) GetPack(ctx context.Context, id uint) (*fleet.Pack, error) {
 
 	return svc.ds.Pack(ctx, id)
 }
+
+////////////////////////////////////////////////////////////////////////////////
+// Create Pack
+////////////////////////////////////////////////////////////////////////////////
+
+type createPackRequest struct {
+	fleet.PackPayload
+}
+
+type createPackResponse struct {
+	Pack packResponse `json:"pack,omitempty"`
+	Err  error        `json:"error,omitempty"`
+}
+
+func (r createPackResponse) error() error { return r.Err }
+
+func createPackEndpoint(ctx context.Context, request interface{}, svc fleet.Service) (interface{}, error) {
+	req := request.(*createPackRequest)
+	pack, err := svc.NewPack(ctx, req.PackPayload)
+	if err != nil {
+		return createPackResponse{Err: err}, nil
+	}
+
+	resp, err := packResponseForPack(ctx, svc, *pack)
+	if err != nil {
+		return createPackResponse{Err: err}, nil
+	}
+
+	return createPackResponse{
+		Pack: *resp,
+	}, nil
+}
+
+func (svc *Service) NewPack(ctx context.Context, p fleet.PackPayload) (*fleet.Pack, error) {
+	if err := svc.authz.Authorize(ctx, &fleet.Pack{}, fleet.ActionWrite); err != nil {
+		return nil, err
+	}
+
+	var pack fleet.Pack
+
+	if p.Name != nil {
+		pack.Name = *p.Name
+	}
+
+	if p.Description != nil {
+		pack.Description = *p.Description
+	}
+
+	if p.Platform != nil {
+		pack.Platform = *p.Platform
+	}
+
+	if p.Disabled != nil {
+		pack.Disabled = *p.Disabled
+	}
+
+	if p.HostIDs != nil {
+		pack.HostIDs = *p.HostIDs
+	}
+
+	if p.LabelIDs != nil {
+		pack.LabelIDs = *p.LabelIDs
+	}
+
+	if p.TeamIDs != nil {
+		pack.TeamIDs = *p.TeamIDs
+	}
```

Comment:
Not for this PR, but I think we can remove this logic here since it's the creation of a Pack, not the update. This logic makes sense for updates so that we know what to change and what to ignore. We should review this pattern and only use it in PATCH endpoints (or update endpoints in general, which could be POST, I suppose).